### ISSUE TITLE
When splitting into train and val partitions, a couple of bugs

### ISSUE THREAD

**Issue by DanielChaseButterfield** [lunarlab-gatech/MorphoSymm-Replication#3]
Two things happen when train and validation partitions are set up:
- The first validation subset has it's first 149 entries ignored, even though you could potentially use the last entries of the train set to be the past history length for the first 149 entries of the validation set.
- Every other validation set uses false history data from other dataset sequences, as previously found in Issue #2 .

**Comment by DanielChaseButterfield** [comment]
Both of these issues were solved with commit dc75c540a1e24fe2e6d449092aed51c3147b79b4 . The second one was solved for the same reasons that Issue #2 was solved. 

The first issue is also resolved for similar reasons. Using `torch.utils.data.Subset` allows us to access the history from the train set, without actually needed to include train set entries in our validation set.

Test cases proving this is working can be found in `tests/test_umich_contact_dataset.py` in the `test_create_train_val_dataset()` function.
